### PR TITLE
Fix/403

### PR DESCRIPTION
**Pull request by blake-regalia**

### CODE REVIEW

File: src/test/kotlin/org/openmbee/mms5/util/Environment.kt [hunk 1]
```diff
@@ -22,22 +22,24 @@ fun testEnv(): ApplicationEngineEnvironment {
 }
 
 
-fun <R> withTest(testName: String, test: TestApplicationEngine.(String) -> R) : R {
+fun <R> withTest(testName: String, test: TestScope?, engine: TestApplicationEngine.(String) -> R) : R {
     return withSystemProperties(mapOf(
         "MMS5_QUERY_URL" to backend.getQueryUrl(),
         "MMS5_UPDATE_URL" to backend.getQueryUrl(),
         "MMS5_GRAPH_STORE_PROTOCOL_URL" to backend.getGspdUrl(),
+        "MMS5_TEST_NO_AUTH" to if(test?.testCase?.config?.tags?.contains(NoAuth) == true) "1" else "",
```
Comment (dlamoris) [inline src/test/kotlin/org/openmbee/mms5/util/Environment.kt:30]: should also test for test == null so it's '1' for the helper functions

File: src/test/kotlin/org/openmbee/mms5/util/Requests.kt [hunk 2]
```diff
@@ -66,6 +68,19 @@ fun TestApplicationRequest.setSparqlQueryBody(body: String) {
 }
 
 fun TestApplicationEngine.httpRequest(method: HttpMethod, uri: String, setup: TestApplicationRequest.() -> Unit): TestApplicationCall {
+    if(true == System.getenv("MMS5_TEST_NO_AUTH")?.isNotBlank()) {
```
Comment (dlamoris) [inline src/test/kotlin/org/openmbee/mms5/util/Requests.kt:71]: shouldn't this be isBlank? if there's a NoAuth tag this env would be 1

**Review by dlamoris — CHANGES_REQUESTED** [review]
there are compile errors, it seems there are some places missing the statuscode pair
from circleci:
e: /application/src/main/kotlin/org/openmbee/mms5/routes/CollectionCreate.kt: (31, 28): Type mismatch: inferred type is String but Pair<String, HttpStatusCode> was expected
e: /application/src/main/kotlin/org/openmbee/mms5/routes/LockCreate.kt: (21, 28): Type mismatch: inferred type is String but Pair<String, HttpStatusCode> was expected
e: /application/src/main/kotlin/org/openmbee/mms5/routes/RepoCreate.kt: (16, 28): Type mismatch: inferred type is String but Pair<String, HttpStatusCode> was expected
e: /application/src/main/kotlin/org/openmbee/mms5/routes/RepoCreate.kt: (30, 28): Type mismatch: inferred type is String but Pair<String, HttpStatusCode> was expected
e: /application/src/main/kotlin/org/openmbee/mms5/routes/RepoCreate.kt: (44, 28): Type mismatch: inferred type is String but Pair<String, HttpStatusCode> was expected
e: /application/src/main/kotlin/org/openmbee/mms5/routes/endpoints/ModelCommit.kt: (152, 37): Type mismatch: inferred type is String but Pair<String, HttpStatusCode> was expected

also i think the env check is still reversed, see comment in code
